python/ngen_conf/src/ngen/config/hydrofabric.py

```diff
@@ -1,74 +1,35 @@
 
 from typing import List, Literal
 from pydantic import BaseModel, validator
-from geojson_pydantic import Feature, Point,  MultiPolygon
+from geojson_pydantic import FeatureCollection, Point,  MultiPolygon
+import re
 
-def check_in_accepted(val, accepted):
-    if val not in accepted:
-        raise ValueError(f'Not an acceptable option. Options are {accepted}')
+# match str that ends with a '-' followed by digits
+_TOID_REGEX_STR = ".\D+-\d+$"
+_TOID_REGEX = re.compile(_TOID_REGEX_STR)
     
-class NexusFeatureProperty(BaseModel):
-    toid: str
-
-    @validator('toid')
-    def check_prefix(cls,val):
-        assert val[:3] == 'cat', "toid must be a cat id"
-
-NexusFeature = Feature[Point, NexusFeatureProperty]
-
 class CatchmentFeatureProperty(BaseModel):
     toid: str
     areasqkm : float
 
     @validator('toid')
-    def check_prefix(cls,val):
-        assert val[:3] == 'nex', "toid must be a nexus id"
-
-CatchmentFeature = Feature[MultiPolygon, CatchmentFeatureProperty]
-
-class CRSProperty(BaseModel):
-    """
-    Model for CRS properties field in catchment or nexus data config
-    """    
-    name: Literal["urn:ogc:def:crs:OGC:1.3:CRS84"]
-
-    # Are any others accepted? If so, check with validator
-    # @validator('name')
-    # def check_properties(cls,val):
-    #     accepted = ["urn:ogc:def:crs:OGC:1.3:CRS84"] # Are any others accepted?
-    #     check_in_accepted(val, accepted)
-
-class CRS(BaseModel):
-    """
-    Model for CRS field in catchment or nexus data config
-    """
-    type: Literal["name"]
-    properties: CRSProperty
-
-    @validator('type')
-    def check_type(cls,val):
-        accepted = ["name"]
-        check_in_accepted(val, accepted)
-
-class NGenCatchmentFile(BaseModel):
-    """
-    Model for catchment data file
-    """
-
-    # required
-    type: Literal["FeatureCollection"]
-    name: Literal["catchment_data"]
-    crs: CRS
-    features: List[CatchmentFeature]
-
+    def _validate_toid(cls, value: str) -> str:
```

Comment:
Sorry, I think this was lost in communication. This function can be moved outside of `CatchmentFeatureProperty` scope. So:

```python

def _validate_toid(value: str) -> str:
    if _TOID_REGEX.search(value) is None:
        raise ValueError(f"Invalid 'toid' property value: {value!r}")
    return value

class CatchmentFeatureProperty(BaseModel):
    toid: str
    areasqkm : float

    _validate_toid = validator("toid", allow_reuse=True)(_validate_toid)


class NexusFeatureProperty(BaseModel):
    toid: str

    _validate_toid = validator("toid", allow_reuse=True)(_validate_toid)
```